src/listpack.c

```diff
@@ -656,7 +656,7 @@ unsigned char *lpInsert(unsigned char *lp, unsigned char *ele, uint32_t size, un
     unsigned char *dst = lp + poff; /* May be updated after reallocation. */
 
     /* Realloc before: we need more room. */
-    if (new_listpack_bytes > old_listpack_bytes) {
+    if (new_listpack_bytes > lp_malloc_size(lp)) {
```

Comment:
```suggestion
    if (new_listpack_bytes > old_listpack_bytes &&
        new_listpack_bytes > lp_malloc_size(lp)) {
```

because `malloc_size()` can be a little bit slow, depending on implementation

Reply:
I think this optimizes for the case of `LP_REPLACE`?